mllib/src/main/scala/org/apache/spark/ml/regression/LinearRegression.scala

```diff
@@ -219,33 +219,41 @@ class LinearRegression @Since("1.3.0") (@Since("1.3.0") override val uid: String
     }
 
     val yMean = ySummarizer.mean(0)
-    val yStd = math.sqrt(ySummarizer.variance(0))
+    var yStd = math.sqrt(ySummarizer.variance(0))
```

Comment:
@srowen I agree with these comments. I'm working on `WeightedLeastSquares` (https://github.com/apache/spark/pull/10274), and there are some commonalities between these two issues. Once I've completed all tests for `WeightedLeastSquares`, I'll compete this issue.

Some of the tests for `LinearRegression` use both `normal` and `l-bfgs` solvers. It would nice if `WeightedLeastSquares` issue is merged so that I can write tests in similar way.